leader/leader.go

```diff
@@ -0,0 +1,35 @@
+package leader
+
+// Status indicates leadership status
+type Status int
+
+const (
+	// StatusNotLeader means the current node is not a leader
+	StatusNotLeader Status = iota
```

Comment:
Following suit with a nit you filed on `UnknownHealth` - the type is already `Status`, so should `Status` be removed from these consts?